source/geometries/Next100FieldCage.cc

```diff
@@ -330,9 +333,16 @@ void Next100FieldCage::BuildActive()
   drift_region->AddRootLogicalVolume(active_logic);
 
   /// Vertex generator
-  active_gen_ = new CylinderPointSampler2020(0., active_ext_radius_, active_length_/2.,
+  active_gen_radius_ =
+    std::max(gate_int_diam_/2. - GetCoordOrigin().x(),
+             gate_int_diam_/2. - GetCoordOrigin().y());
+
+  active_gen_radius_ = std::max(active_gen_radius_, active_ext_radius_);
```

Comment:
The thing is that the active volume is made of the union of two pieces, a polyhedron and a cylinder. On top of that, these pieces are not glued centered on the same axis, but displaced due to the field cage displacement. Therefore, depending on the value of the displacement, it can be that one needs a bigger radius for the generator, than the external radius of the polyhedron. Notice that the generator is a cylinder that needs to overcover all the active volume, to keep the code simple.